### ISSUE TITLE
PythonEngine.Shutdown() Crash

### ISSUE THREAD

**Issue by nlogozzo** [pythonnet/pythonnet#2107]
### Environment

-   Pythonnet version: 3.0.1
-   Python version: 3.7.3
-   Operating System: Windows 11 (Same happens when running on WSL - Ubuntu)
-   .NET Runtime: .NET 7

### Details

Calling `Python.Runtime.PythonEngine.Shutdown();` results in the error:
```
System.NotSupportedException: 'BinaryFormatter serialization and deserialization are disabled within this application. See https://aka.ms/binaryformatter for more information.'
```

**Comment by filmor** [comment]
Workaround here: https://github.com/pythonnet/pythonnet/issues/2008#issuecomment-1441315835

**Comment by lostmsu** [comment]
@filmor do we even need `Shutdown` on .NET Core?

**Comment by filmor** [comment]
We need to call `Shutdown` (in my opinion) just as well to ensure that destructors run in a sensible order. What we definitely don't need to do is the stashing part. It only needs to run if we want to reload domains, which is not supported on .NET Core anyway. I'm currently working on this (and related slowness during shutdowns) here:

#2245

**Comment by JKingas** [comment]
I recently tried pythonnet in C# Console Application in Visual Studio 2022.

**Environment**
- Pythonnet version: 3.0.3
- Python version: 3.12.3
- Operating System: Windows 11 23H2
- .NET Runtime: .NET 8

For now, `Shutdown` is needed, because without it, one of the threads does not allow the process to end completely.
Then I tried to call `Shutdown` but got this exception in the PR.

In this case, need to add `<EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>` to the project file. Even with this flag, the process does not terminate without `Shutdown`.

And after this flag and call `Shutdown` the process ends correctly.
